### PR TITLE
glider 0.16.2 (new formula)

### PR DESCRIPTION
**Pull request by icpz**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

glider is a powerful forward proxy https://github.com/nadoo/glider

### CODE REVIEW

File: Formula/glider.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  url "https://github.com/nadoo/glider/archive/refs/tags/v0.16.2.tar.gz"
+  sha256 "a1c7032ad508b6c55dad3a356737cf05083441ea16a46c03f8548d4892ff9183"
+  license "GPL-3.0-or-later"
+  head "https://github.com/nadoo/glider.git"
+
+  depends_on "go" => :build
+
+  on_linux do
+    depends_on "curl" => :test
+  end
```
Comment (SMillerDev) [inline Formula/glider.rb:13]:
```suggestion
  uses_from_macos "curl" => :test
```

Comment (icpz) [reply Formula/glider.rb]: Thanks!

File: Formula/glider.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+class Glider < Formula
+  desc "Forward proxy with multiple protocols support"
+  homepage "https://github.com/nadoo/glider"
+  url "https://github.com/nadoo/glider/archive/refs/tags/v0.16.2.tar.gz"
+  sha256 "a1c7032ad508b6c55dad3a356737cf05083441ea16a46c03f8548d4892ff9183"
+  license "GPL-3.0-or-later"
+  head "https://github.com/nadoo/glider.git"
```
[the remark this answers is not in the document]
Comment (icpz) [reply Formula/glider.rb]: The branch should be `dev`, thanks! I've made the change.

[the remark this answers is not in the document]
Comment (icpz) [reply Formula/glider.rb]: Emmm... The CI seems to fail with message
```
  Error: 1 problem in 1 formula detected
  Error: HEAD: Use `branch: "master"` to specify the default branch`
```

How to specify a non default branch for head build?

[the remark this answers is not in the document]
Comment (chenrui333) [reply Formula/glider.rb]: you should change the default branch on the github repo side.

[the remark this answers is not in the document]
Comment (chenrui333) [reply Formula/glider.rb]: there is some auditing logic in place to check this.

[the remark this answers is not in the document]
Comment (icpz) [reply Formula/glider.rb]: Thanks, I've change the branch, now the CI has passed

File: Formula/glider.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  end
+
+  test do
+    proxy_port = free_port
+    filename = (testpath/"test.tar.gz")
+    glider = fork { exec "#{bin}/glider", "-listen", "socks5://:#{proxy_port}" }
+
+    sleep 3
+    begin
+      system "curl", "--socks5", "127.0.0.1:#{proxy_port}", "-L", stable.url, "-o", filename
```
Comment (SMillerDev) [inline Formula/glider.rb:35]: Let's not use the internal Homebrew APIs here in the test.
```suggestion
      system "curl", "--socks5", "127.0.0.1:#{proxy_port}", "-L", "https://brew.sh", "-o", filename
```

Comment (icpz) [reply Formula/glider.rb]: Done, thanks!

**Review by chenrui333 — APPROVED** [review]

**Comment by nadoo** [comment]
Thanks!

**Pull request closed without merging**